astropy/modeling/tests/test_compound.py

```diff
@@ -543,175 +282,72 @@ def test_identity_input():
-
     with pytest.raises(ModelDefinitionError):
-        # The slice can't actually be taken since the resulting model cannot be
-        # evaluated
-        assert m[1:].submodel_names == ('b', 'c', 'd', 'e')
-
-    assert m[:].submodel_names == ('a', 'b', 'c', 'd', 'e')
-    assert m['a':].submodel_names == ('a', 'b', 'c', 'd', 'e')
-
-    with pytest.raises(ModelDefinitionError):
-        assert m['c':'d'].submodel_names == ('c', 'd')
-
-    assert m[1:2].name == 'b'
-    assert m[2:7].submodel_names == ('c', 'd', 'e')
-    with pytest.raises(IndexError):
-        m['x']
-    with pytest.raises(IndexError):
-        m['a': 'r']
-
-    with pytest.raises(ModelDefinitionError):
-        assert m[-4:4].submodel_names == ('b', 'c', 'd')
+        Rotation2D(90) | Gaussian1D(1, 0, 0.1)
 
     with pytest.raises(ModelDefinitionError):
-        assert m[-4:-2].submodel_names == ('b', 'c')
-
-
-def test_slicing_on_instances_3():
-    """
-    Like `test_slicing_on_instances_2` but uses a compound model that does not
-    have any invalid slices due to the resulting model being invalid
-    (originally test_slicing_on_instances_2 passed without any
-    ModelDefinitionErrors being raised, but that was before we prevented
-    invalid models from being created).
-    """
-
-    model_a = Shift(1, name='a')
-    model_b = Shift(2, name='b')
-    model_c = Gaussian1D(3, 0, 0.1, name='c')
-    model_d = Scale(2, name='d')
-    model_e = Scale(3, name='e')
-
-    m = (model_a + model_b) | model_c | (model_d + model_e)
-
-    assert m[1:].submodel_names == ('b', 'c', 'd', 'e')
-    assert m[:].submodel_names == ('a', 'b', 'c', 'd', 'e')
-    assert m['a':].submodel_names == ('a', 'b', 'c', 'd', 'e')
-    assert m['c':'d'].submodel_names == ('c', 'd')
-    assert m[1:2].name == 'b'
-    assert m[2:7].submodel_names == ('c', 'd', 'e')
-    with pytest.raises(IndexError):
-        m['x']
-    with pytest.raises(IndexError):
-        m['a': 'r']
-    assert m[-4:4].submodel_names == ('b', 'c', 'd')
-    assert m[-4:-2].submodel_names == ('b', 'c')
-
-
-def test_slicing_on_instance_with_parameterless_model():
-    """
-    Regression test to fix an issue where the indices attached to parameter
-    names on a compound model were not handled properly when one or more
-    submodels have no parameters.  This was especially evident in slicing.
-    """
-
-    p2 = Polynomial2D(1, c0_0=1, c1_0=2, c0_1=3)
-    p1 = Polynomial2D(1, c0_0=1, c1_0=2, c0_1=3)
-    mapping = Mapping((0, 1, 0, 1))
-    offx = Shift(-2, name='x_translation')
-    offy = Shift(-1, name='y_translation')
-    aff = AffineTransformation2D(matrix=[[1, 2], [3, 4]], name='rotation')
-    model = mapping | (p1 & p2) | (offx & offy) | aff
-
-    assert model.param_names == ('c0_0_1', 'c1_0_1', 'c0_1_1',
-                                 'c0_0_2', 'c1_0_2', 'c0_1_2',
-                                 'offset_3', 'offset_4',
-                                 'matrix_5', 'translation_5')
```

Comment:
This fails because `model.param_names` returns now a list while previously it returned a tuple. Do we want `param_names` to be mutable?